diff -uNdr deadbeef-0.7.2_orig/plugins/adplug/adplug/s3m.cpp deadbeef-0.7.2_new/plugins/adplug/adplug/s3m.cpp
--- deadbeef-0.7.2_orig/plugins/adplug/adplug/s3m.cpp	2016-06-19 14:26:18.000000000 +0300
+++ deadbeef-0.7.2_new/plugins/adplug/adplug/s3m.cpp	2018-05-10 14:08:22.481231976 +0300
@@ -26,7 +26,7 @@
 #include <string.h>
 #include "s3m.h"
 
-const char Cs3mPlayer::chnresolv[] =	// S3M -> adlib channel conversion
+const int8_t Cs3mPlayer::chnresolv[] =	// S3M -> adlib channel conversion
   {-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,0,1,2,3,4,5,6,7,8,-1,-1,-1,-1,-1,-1,-1};
 
 const unsigned short Cs3mPlayer::notetable[12] =		// S3M adlib note table
diff -uNdr deadbeef-0.7.2_orig/plugins/adplug/adplug/s3m.h deadbeef-0.7.2_new/plugins/adplug/adplug/s3m.h
--- deadbeef-0.7.2_orig/plugins/adplug/adplug/s3m.h	2016-06-19 14:26:18.000000000 +0300
+++ deadbeef-0.7.2_new/plugins/adplug/adplug/s3m.h	2018-05-10 14:07:39.071591002 +0300
@@ -23,6 +23,7 @@
 #define H_ADPLUG_S3M
 
 #include "player.h"
+#include <cstdint>
 
 class Cs3mPlayer: public CPlayer
 {
@@ -92,7 +93,7 @@
   char filetype[30];
 
  private:
-  static const char chnresolv[];
+  static const int8_t chnresolv[];
   static const unsigned short notetable[12];
   static const unsigned char vibratotab[32];
 
